subprojects/libvhost-user/libvhost-user.c

```diff
@@ -1397,6 +1397,44 @@ bool vu_set_queue_host_notifier(VuDev *dev, VuVirtq *vq, int fd,
     return vu_process_message_reply(dev, &vmsg);
 }
 
+bool vu_get_shared_object(VuDev *dev, unsigned char uuid[16], int *dmabuf_fd)
+{
+    int result = false;
+    VhostUserMsg msg_reply;
+    VhostUserMsg msg = {
+        .request = VHOST_USER_BACKEND_SHARED_OBJECT,
+        .size = sizeof(msg.payload.object),
+        .flags = VHOST_USER_VERSION | VHOST_USER_NEED_REPLY_MASK,
+        .payload.object = {
+            .type = VHOST_SHARED_OBJECT_LOOKUP,
+        },
+    };
+
+    memcpy(msg.payload.object.uuid, uuid, sizeof(uuid[0]) * 16);
```

Comment:
It is more in case the type changes, to avoid simple programming errors. I think being more verbose and showing here where the size comes from is better.